apps/hannk/interpreter/model.h

```diff
@@ -349,30 +337,43 @@ class OpGroup : public Op {
         : Op(std::move(inputs), std::move(outputs)), ops_(std::move(ops)) {
     }
 
-    void add(OpPtr to_insert, const Op *before = nullptr);
-    void remove(const Op *op);
-
     BoundsMap map_bounds(int input_idx, int output_idx) const override;
 
     bool prepare() override;
-
     void execute() override;
 
     int op_count() const {
         return ops_.size();
     }
+    OpPtr op_ptr(int i) {
+        OpPtr result = nullptr;
+        std::swap(ops_[i], result);
```

Comment:
Yep -- this is used by `OpMutator::visit` to mutate an OpGroup.  It needs to take ownership of the unique_ptr so it can grant ownership to the new OpGroup. (The implementation with `swap` is probably overkill; I wanted to be sure I wasn't leaving the old OpGroup in an inconsistent state, but that's probably not essential.)

Reply:
Should this be renamed then? `take_op` or something?

Reply:
Done